testing/MLDB-1235-temporal-aggregators.py

```diff
@@ -439,36 +439,37 @@ def test_mldbfb_344_temporal_segfault(self):
                 ]
             )
 
-    @unittest.expectedFailure
     def test_mldbfb_516_aggregator_incorrect_with_join(self):
         ds = mldb.create_dataset({
-            'id' : 'ds_join',
-            'type' : 'beh.binary.mutable'
+            'id' : 'ds516',
+            'type' : 'sparse.mutable'
         })
         ds.record_row('user3', [['behA', 1, 11], ['conv', 1, 70],
                                 ['behB', 1, 14], ['behA', 1, 14]])
         ds.commit()
 
         ds = mldb.create_dataset({
             'id' : 'conv',
-            'type' : 'beh.mutable'
+            'type' : 'sparse.mutable'
         })
         ds.record_row('user3', [['ts', 70, 0]])
         ds.commit()
 
         res = mldb.query("""
-            SELECT temporal_count({ds_join.*}) AS *
-            FROM ds_join
+            SELECT temporal_count({ds516.*}) AS *
+            FROM ds516
         """)
+        mldb.log(res)
         self.assertTableResultEquals(res, [
             ['_rowName', 'behA', 'behB', 'conv'],
             ['user3', 2, 1, 1]
         ])
 
         res = mldb.query("""
-            SELECT temporal_count({ds_join.*}) AS *
-            FROM ds_join INNER JOIN conv ON ds_join.rowName() = conv.rowName()
+            SELECT temporal_count({ds516.* as *}) AS *
```

Comment:
That's a long-standing and deeper (but separate) thing.  In the context of a single dataset `x` with columns `a=1` and `b=2`, `select * from x` returns `{ a: 1, b: 2}`, not `{ x: { a: 1, b: 2} }`.   When the support for select `x.* from x` was added, it was done in such a way that the superfluous prefix was detected and simply removed; so that it gives the same result.  In the context of a join, however, the table name is always added.

I would say that choosing and implementing a consistent solution (that's consistent with the standard, or at least with other SQL implementations) is a separate issue.